interface/interface.c

```diff
@@ -353,11 +359,16 @@ check_for_new_dps(void)
 {
 	struct in_addr a;
 	uint32_t addr;
+	uint64_t msg_bundle;
 	memset(&addr, 0, sizeof(addr));
+	msg_bundle = 0;
 	/* receive request */
-	int n = zmq_recv(dp_sock, &addr, sizeof(addr), 0);
+	int n = zmq_recv(dp_sock, &msg_bundle, sizeof(msg_bundle), 0);
 	assert(n != -1);
+	/* DP's IP in the first 32 B; S1U's IP address to the last 32 B */
+	addr = (uint32_t)(msg_bundle >> (sizeof(addr) * CHAR_BIT));
 	a.s_addr = addr;
+	s1u_sgw_ip.s_addr = (msg_bundle & 0xFFFFFFFF);
```

Comment:
@ajamshed This does not seem to be per DP?

Reply:
Not yet. This will eventually go in, _e.g._ ``ue_context`` struct once #90 is merged. Currently this patch makes the CP's run-time argument of passing S1U_SGW_IP redundant.